packages/db/src/search/GranuleSearch.ts

```diff
@@ -116,9 +119,30 @@ export class GranuleSearch extends BaseSearch {
    * @param pgRecords - postgres records returned from query
    * @returns translated api records
    */
-  protected translatePostgresRecordsToApiRecords(pgRecords: GranuleRecord[])
-    : Partial<ApiGranuleRecord>[] {
+  protected async translatePostgresRecordsToApiRecords(pgRecords: GranuleRecord[], knex: Knex)
+    : Promise<Partial<ApiGranuleRecord>[]> {
     log.debug(`translatePostgresRecordsToApiRecords number of records ${pgRecords.length} `);
+
+    const cumulusIds = pgRecords.map((record) => record.cumulus_id);
+
+    //get Files
+    const fileModel = new FilePgModel();
+    const files = await fileModel.searchByGranuleCumulusIds(knex, cumulusIds);
```

Comment:
🐜 :  This likely winds up being a memory multiplier, while batching is generally more performant versus a ton of connections for bulk queries, we do need to keep in mind in-memory limits when using this pattern. 

More relevant to this PR: is this actually just a Search method?      We should consider if this should be protected and bound to ES style searches given we should adopt better bulk query approaches elsewhere in the codebase.     

We generally should not be using the search scaffolding for internal queries.

Reply:
memory optimized significantly so it is not a *multiplier* although we need to go back to the drawing board if this bulk query is too memory intensive